qcog_python_client/qcog/_baseclient.py

```diff
@@ -299,15 +327,101 @@ async def _preloaded_training_parameters(self, guid: str) -> BaseQcogClient:
         )
         return self
 
-    async def _preloaded_model(self, guid: str) -> BaseQcogClient:
+    async def _preloaded_model(
+        self,
+        guid: str | None = None,
+        *,
+        pytorch_model_name: str | None = None,
+        force_reload: bool = False,
+    ) -> BaseQcogClient:
         """Retrieve preexisting model payload."""
+        # If a `pytorch_model_name` is provided,
+        # We can assume that we don't have a `model`
+        # set yet, so we will fetch the model first.
+        if pytorch_model_name:
+            await self._preloaded_pt_model(pytorch_model_name)
+
         if self.model.model_name == Model.pytorch.value:
-            pytorch_model_guid = self.pytorch_model["guid"]
+            await self._preload_trained_pt_model(
+                guid=guid,
+                force_reload=force_reload,
+            )
+        else:
+            if guid is None:
+                raise ValueError(
+                    "Model guid is required for Pauli and Ensemble models."
+                )
+            await self._preload_trained_qcog_model(guid)
+
+        return self
+
+    async def _preload_trained_pt_models(
+        self,
+        pytorch_model_guid: str,
+        *,
+        page: int = 0,
+        limit: int = 100,
+        training_status: TrainingStatus | None = None,
+    ) -> BaseQcogClient:
+        """Retrieve preexisting trained models for a PyTorch model."""
+        params: dict = {
+            "limit": limit,
+            "page": page,
+        }
+
+        if training_status:
+            params["training_status"] = training_status.value
+
+        # Compose URL with parameters
+        # TODO: refactor methods to accept parameters dictionary
+        # and move this logic to the http client
+        url = f"pytorch_model/{pytorch_model_guid}/trained_model"
+        urlparts = urlparse(url)
+        query_params = parse_qs(urlparts.query)
+        query_params.update(params)
+        new_query_string = urlencode(query_params, doseq=True)
+        new_url_parts = urlparts._replace(query=new_query_string)
+        url = urlunparse(new_url_parts)
+
+        self.pytorch_trained_models = await self.http_client.get_many(
+            url,
+        )
+        return self
+
+    async def _preload_trained_pt_model(
+        self,
+        *,
+        guid: str | None = None,
+        force_reload: bool = False,
+    ) -> BaseQcogClient:
+        # If a guid is provided, we will fetch the trained model
+        pytorch_model_guid = self.pytorch_model["guid"]
+
+        if guid:
             self.trained_model = await self.http_client.get(
                 f"pytorch_model/{pytorch_model_guid}/trained_model/{guid}"
             )
-        else:
-            self.trained_model = await self.http_client.get(f"model/{guid}")
+
+            return self
+        # Otherwise, check if we need to load the latest trained models
+        if force_reload:
```

Comment:
elif? These should be exclusive right we wouldn't load the GUID and then throw that model away and load latest with the force reload?  Probably an error to include both?

Reply:
the `return` should break the code here, so if it executes the first if it doesn't check for the second.